### ISSUE TITLE
HV cable not reliably doing damage with nano armor

### ISSUE THREAD

**Issue by Joomzy** [TinyModularThings/IC2Classic#321]
I read that the HV cable is supposed to also count as a electric fence. When testing this out I got some weird results. 

Upon testing it with a full set of Diamond Armor, I took damage and it took durability on the armor set.

When using Nano or QuantumSuit armor it did no damage unless standing on a particular spot on the MFSU. It also did no energy drain on the armor set.

Video down below.
https://www.youtube.com/watch?v=f_Ptf1AuBTg

**Comment by Speiger** [comment]
@Joomzy this made me laught really hard.
Question: Electric Armor + Electric Damage, what do you think is going to happen? The Damage gets absorbed and actually charge the armor. Guess what happens if you get struct by lightning (Avengers shows that even with a joke xD)

Not a bug, i actually coded it that way.
Also video is not watchable

**Comment by Joomzy** [comment]
Sure, the durability makes sense, but if you'd watch the video then it shows why I pointed this out. I'll check why it's not watchable. I wasn't only talking about durability.

**Comment by Speiger** [comment]
The logic which absorbs the EU does is in the armor so even if it does 0Damage the damage animation is enforced nothing i can really do about it as of my current knowledge. (Not that I have researched that)

The MFSU thing is actually intentional, Every cable checks a full 3x3x3 cube around itself if entities collide with it.
Else the calculation would be way to complex. Also since mc has no sense of rotation by default collision checks I would have to implement a complex System that accounts every case.

TL:DR Its a crude implementation that is ment to just work and not account for every unique case.
Also the Enet needs to stay performant some how since it manages the entirety of electric flow.

**Comment by Joomzy** [comment]
Makes sense, was just weird how it did damage to me only on that corner of the MFSU and not anywhere else.

**Comment by Speiger** [comment]
I just saw what you meant staying on the edge of causes damage.
That is actually a unique counter measure against abusing this system and not making it to overpowered.
If the armor is full of charge and can not absorb the electric charge it just applies the damage directly since its no longer able to protect you from that, unless you are wearing electric boots & stay on solid ground. (grounding yourself)
Minecrafts detection system on "Ground Detection" is not the best. So staning on a edge like that counts staying in air. So thats why you are taking damage.

**Comment by Speiger** [comment]
just in case now a ping @Joomzy
